### PR TITLE
Feat/cache CashCtrlClient class

### PR DESCRIPTION
**Pull request by AlexTheWizardL**
This pull request covers caching for `CashCtrlClient` class that is implemented in `CachedCashCtrlClient` class.

@lasuk Please review the changes and provide feedback or approval for merging.

### CODE REVIEW

File: cashctrl_api/cashed_client.py [hunk 1]
```diff
@@ -64,120 +72,302 @@ def list_accounts(self) -> pd.DataFrame:
...
+        self._journal_entries_cache_time = None
+
+    def invalidate_files_cache(self) -> None:
+        """
+        Invalidates the cached files data.
+        """
+        self._files_cache = None
+        self._files_cache_time = None
+
     def _is_expired(self, cache_time: Optional[datetime]) -> bool:
```
Comment (lasuk) [inline cashctrl_api/cashed_client.py:384]: I personally prefer `datetime | None` over `Optional[datetime]`.
(Requires a newer python version, though, which is fine for us.)

File: tests/test_cached_account_categories.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+def test_account_category_to_id(cc_client, account_categories):
+    assert cc_client.account_category_to_id(account_categories['path'].iat[1]) == account_categories['id'].iat[1], (
+        'Cached account category id doesn`t correspond actual id'
+    )
+
+def test_account_category_to_id_with_invalid_account_category_raises_error(cc_client):
+    with pytest.raises(ValueError, match='No id found for account category path'):
+        cc_client.account_category_to_id(99999999)
+
```
Comment (lasuk) [inline tests/test_cached_account_categories.py:43]: We should also test that `account_category_[to|from]_id(99999999, allow_missing=True)` returns None.

(Also applies to other mappers with an `allow_missing` argument.)

**Review by lasuk — COMMENTED** [review]
Thanks a lot for extending the cached client and for adding test cases.

I've pushed commit b3f2cf848 to enhance documentation and code readability and left a few comments.
@AlexTheWizardL, please shout if you disagree.

File: tests/test_cached_account_categories.py (hunk 2 above)
Comment (AlexTheWizardL) [reply tests/test_cached_account_categories.py:43]: I will add this for tax rates and accounts
